client/rest/src/main/java/org/elasticsearch/client/RestClient.java

```diff
@@ -434,10 +434,10 @@ private void setHeaders(HttpRequest httpRequest, Header[] requestHeaders) {
         for (Header requestHeader : requestHeaders) {
             Objects.requireNonNull(requestHeader, "request header must not be null");
             httpRequest.addHeader(requestHeader);
-            requestNames.add(requestHeader.getName());
+            requestNames.add(requestHeader.getName().toLowerCase(Locale.ENGLISH));
         }
         for (Header defaultHeader : defaultHeaders) {
-            if (requestNames.contains(defaultHeader.getName()) == false) {
+            if (!requestNames.contains(defaultHeader.getName().toLowerCase(Locale.ENGLISH))) {
```

Comment:
Like it or not, our style is to use ` == false` instead of `!`.

Reply:
I’ll change it back then!